
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>4.1. Introduction &#8212; Phenotype from Genotype</title>
    
  <link href="../_static/css/theme.css" rel="stylesheet">
  <link href="../_static/css/index.ff1ffe594081f20da1ef19478df9384b.css" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      

    
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/sphinx-book-theme.css?digest=c3fdc42140077d1ad13ad2f1588a4309" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../_static/js/index.be7d3bbb2ef33a8344ce.js">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/sphinx-book-theme.12a9622fbb08dcb3a2a40b2c02b83a57.js"></script>
    <script async="async" src="https://unpkg.com/thebe@0.5.1/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="4.2. Snowflake Algorithm" href="2-snowflake-algorithm.html" />
    <link rel="prev" title="4. Phenotype prediction with Snowflake" href="0-index.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/03-abstract.html">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../c0-front-matter/05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
            <!-- Source interaction buttons -->

            <!-- Full screen (wrap in <a> to have style consistency -->

<a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
        data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
        title="Fullscreen mode"><i
            class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
            <div class="tocsection onthispage pt-5 pb-3">
                <i class="fas fa-list"></i> Contents
            </div>
            <nav id="bd-toc-nav" aria-label="Page">
                <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#motivation">
   4.1.1. Motivation
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#related-work">
   4.1.2. Related work
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#phenotype-predictors-and-variant-prioritisation">
     4.1.2.1. Phenotype predictors and variant prioritisation
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#clustering-and-outlier-detection-in-genetics">
     4.1.2.2. Clustering and outlier-detection in genetics
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#overcoming-the-curse-of-dimensionality-through-dimensionality-reduction-and-feature-selection">
     4.1.2.3. Overcoming the curse of dimensionality through dimensionality reduction and feature selection
    </a>
   </li>
  </ul>
 </li>
</ul>

            </nav>
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="introduction">
<span id="snowlflake-intro"></span><h1><span class="section-number">4.1. </span>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h1>
<p>The Snowflake algorithm is primarily a phenotype prediction method: it takes data about individuals DNA as input and outputs predictions about which phenotypes each individual has.
These predictions are based on how unusual an individual is for variants relating to each phenotype, and are made across a breadth of phenotypes and for missense variants across the protein-coding genome.
It does this by combining existing predictions of variant deleteriousness from FATHMM<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id53"><span>113</span></a>]</span> and association of protein domains to phenotypes from DcGO<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id51"><span>109</span></a>]</span>, and finding unusual combinations of these variants through clustering individuals against a diverse background cohort and looking for outliers.
The phenotype prediction implicitly contains protein function predictions, due to <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html#phenotype-protein-prediction-link"><span class="std std-ref">the relationship between protein function and phenotype</span></a>, and these are the key output of Snowflake.
As a protein function predictor, Snowflake seeks rare combinations of SNPs which may influence a phenotype.
In other words, Snowflake looks for the mechanisms behind complex traits, which are currently not well understood, but are thought to cause many human diseases.</p>
<div class="section" id="motivation">
<span id="snowflake-intro-motivation"></span><h2><span class="section-number">4.1.1. </span>Motivation<a class="headerlink" href="#motivation" title="Permalink to this headline">¶</a></h2>
<p>In <a class="reference internal" href="../c02-biology-background/0-index.html#c02-biology-bg"><span class="std std-numref">chapter 2</span></a>, we discussed the theoretical mechanism from which phenotype arises from genotype.
In summary: differences in DNA cause differences in cell functionality, which interact with the cell environment to create differences in overall phenotype.</p>
<p>As <a class="reference internal" href="../c02-biology-background/5-phenotype.html#which-phenotypes"><span class="std std-ref">previously mentioned</span></a>, many recognised phenotypes are medical disorders or their symptoms.
Currently to achieve diagnoses for genetic illnesses, specific genes are often sequenced one at a time, since looking at whole genomes would be too time consuming for clinical staff.
Patients seeking diagnoses for rare genetic diseases describe the process as an “Odyssey”, more than half undiagnosed at any given time.
If whole genome (or genotype) based phenotype prediction was possible, only one sample and test would be needed to get a much fuller picture of a person’s health, and we would be able to reduce the long and tiring process of obtaining diagnoses for rare genetic diseases.
Applied to the plant and animal kingdom, phenotype prediction could also be beneficial in veterinary science and agriculture.</p>
<p>The discovery of underlying mechanisms for complex traits remains a particular challenge.</p>
</div>
<div class="section" id="related-work">
<span id="snowflake-related-work"></span><h2><span class="section-number">4.1.2. </span>Related work<a class="headerlink" href="#related-work" title="Permalink to this headline">¶</a></h2>
<div class="section" id="phenotype-predictors-and-variant-prioritisation">
<span id="phenotype-predictors-variant-prioritisation"></span><h3><span class="section-number">4.1.2.1. </span>Phenotype predictors and variant prioritisation<a class="headerlink" href="#phenotype-predictors-and-variant-prioritisation" title="Permalink to this headline">¶</a></h3>
<p>Biology databases are home curated, open data that cover genomes across the tree of life, as well as cross-species ontologies of biological processes, diseases, and anatomical entities.
There are a number of recent phenotype prediction methods that have had some success in using these resources for either variant prioritisation or use as a clinical diagnostic tool.</p>
<p>There are a class of “knowledge-based” methods, which use knowledge from databases of experimental results (known associations between genes and phenotypes) as the basis for these predictions, for example Phen-Gen<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id252"><span>155</span></a>]</span>, dcGO<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id51"><span>109</span></a>]</span>, PhenoDigm<span id="id5">[<a class="reference internal" href="../cz-end-matter/reference.html#id257"><span>156</span></a>]</span>, and PHIVE<span id="id6">[<a class="reference internal" href="../cz-end-matter/reference.html#id254"><span>157</span></a>]</span>.
The better performing methods in this class, use associations between model organisms and orthologous genes, to leverage the wealth of information that is collected from these model organism experiments.</p>
<p>There are also “functional” methods, like FATHMM<span id="id7">[<a class="reference internal" href="../cz-end-matter/reference.html#id53"><span>113</span></a>]</span> and CADD<span id="id8">[<a class="reference internal" href="../cz-end-matter/reference.html#id256"><span>158</span></a>]</span>, which instead use information about how the molecules and their function may change with different nucleotide or amino acid substitutions, as well as conservation metrics to prioritise variants.
These tools rank variants for deleteriousness, but do not link them to specific phenotypes.</p>
<p>Most successful methods of any kind now combine multiple sources of information, some combine both functional and knowledge-based sources.
This approach is used within Exomiser<span id="id9">[<a class="reference internal" href="../cz-end-matter/reference.html#id255"><span>159</span></a>]</span>, which combines PHIVE with many other sources of information such as protein-protein interactions, cross-species phenotype associations, and variant frequency using a black-box classifier.
Phenolyzer<span id="id10">[<a class="reference internal" href="../cz-end-matter/reference.html#id253"><span>160</span></a>]</span> and Genomiser<span id="id11">[<a class="reference internal" href="../cz-end-matter/reference.html#id251"><span>161</span></a>]</span> also take similar approaches of combining many different sources of data.</p>
<p>The aim of these models is mostly to prioritise variants associated with diseases, and they are bench-marked by their ability to identify known variants.
Lists of known variants may be purpose-curated from the literature according to specific evidence, or may come from some subset of annotation databases (which in some cases the algorithm may have used as input data).
Each phenotype predictor often targets a specific use case (e.g. non-coding variants), and in combination with the varying validation methods used, it is difficult to compare the accuracy of all of these models directly.
For this reason, the CAFA competition is very useful in getting a more objective view of the capabilities of these kinds of tools.</p>
<p>Similar approaches have been used as clinical diagnostic tools.
PhenIX (Phenotypic Interpretation of eXomes)<span id="id12">[<a class="reference internal" href="../cz-end-matter/reference.html#id250"><span>162</span></a>]</span> is a version of PHIVE which is restricted to the “human disease-causing genome” (genes known to cause disease) to make it more suitable for clinical use in diagnosis of rare genetic diseases.
It also includes semantic similarity information between inputted symptoms and Human Phenotype Ontology terms, using the Phenomizer<span id="id13">[<a class="reference internal" href="../cz-end-matter/reference.html#id249"><span>163</span></a>]</span> algorithms.
For PhenIX the measure of success is that it enabled skilled clinicians to find diagnoses for 11 out of 40 (28%) patients with rare genetic diseases, who were not able to be diagnosed through other means.</p>
<p>While these examples are the most similar published work to Snowflake, they are all tested as variant prioritisation tools rather than phenotype predictors.</p>
</div>
<div class="section" id="clustering-and-outlier-detection-in-genetics">
<span id="clustering-outlier-genetics"></span><h3><span class="section-number">4.1.2.2. </span>Clustering and outlier-detection in genetics<a class="headerlink" href="#clustering-and-outlier-detection-in-genetics" title="Permalink to this headline">¶</a></h3>
<div class="margin sidebar" id="haplotypes">
<p class="sidebar-title">Haplotypes and haplogroups</p>
<p>Haplotypes are groups of alleles that are inherited together from a single parent.
In humans, Y-chromosomes and mitochondrial DNA (mtDNA) are often analysed for haplotypes to understand ancestry within species since these are passed from parents to children without recombination (only mothers pass on mtDNA, and only fathers pass on Y chromosomes).</p>
<p>Haplogroups are groups of haplotypes, representing major branching points in the within-species phylogenetic tree.</p>
</div>
<p>Clustering algorithms, particularly hierarchical methods, are commonly used in genetics for:</p>
<ol class="simple">
<li><p>finding evolutionary relationships between DNA samples, for example, in reconstructing phylogenetic trees and mapping <a class="reference internal" href="#haplotypes"><span class="std std-ref">haplotypes</span></a> within populations<span id="id14">[<a class="reference internal" href="../cz-end-matter/reference.html#id268"><span>164</span></a>]</span>.</p></li>
<li><p>finding functional relationships between genes based on gene expression data<span id="id15">[<a class="reference internal" href="../cz-end-matter/reference.html#id267"><span>165</span></a>,<a class="reference internal" href="../cz-end-matter/reference.html#id270"><span>166</span></a>]</span>.</p></li>
</ol>
<p>For applications in group (1), individuals are generally separated into clusters based on their DNA variants, whereas for (2) samples are separated into clusters based on their gene expression.</p>
<p>Clustering methods are only very rarely used to cluster individuals in phenotype prediction or variant prioritisation tasks.
In one case, clustering individuals based on a combination of genotype and phenotype information has been applied to identify subtypes within emphysema<span id="id16">[<a class="reference internal" href="../cz-end-matter/reference.html#id269"><span>167</span></a>]</span> (a lung disease).</p>
</div>
<div class="section" id="overcoming-the-curse-of-dimensionality-through-dimensionality-reduction-and-feature-selection">
<span id="curse-of-dimensionality-section"></span><h3><span class="section-number">4.1.2.3. </span>Overcoming the curse of dimensionality through dimensionality reduction and feature selection<a class="headerlink" href="#overcoming-the-curse-of-dimensionality-through-dimensionality-reduction-and-feature-selection" title="Permalink to this headline">¶</a></h3>
<div class="figure align-default" id="curse-of-dimensionality">
<a class="reference internal image-reference" href="../_images/curse-of-dimensionality.png"><img alt="../_images/curse-of-dimensionality.png" src="../_images/curse-of-dimensionality.png" style="height: 220px;" /></a>
<p class="caption"><span class="caption-number">Fig. 4.1 </span><span class="caption-text">As the number of dimensions increase from 1 dimension on the left to 3 on the right, the number of points needed to cover the space increases exponentially. This means that for a fixed number of points (individuals), increasing the number of dimensions (SNPs) that we cluster on decreases the density of our space exponentially, making it exponentially harder to identify clusters.</span><a class="headerlink" href="#curse-of-dimensionality" title="Permalink to this image">¶</a></p>
</div>
<p>The “curse of dimensionality” is a phrase coined by Richard E Bellman, during his work on dynamic programming<span id="id17">[<a class="reference internal" href="../cz-end-matter/reference.html#id124"><span>168</span></a>]</span>, but has since proved relevant in many different mathematical and data-driven fields.
While it’s used colloquially as a catch-all complaint about high-dimensional data, the “curse” specifically refers to the sparsity of data that occurs exponentially with an increase in dimensions.
This leads to various problems in different fields<span id="id18">[<a class="reference internal" href="../cz-end-matter/reference.html#id125"><span>169</span></a>]</span> including general difficulty in reaching statistical significance and reduced usefulness of clustering, distance, and outlier metrics.
This can easily be a problem in genetics since we have tens of thousands of genes, and hundreds of thousands of variants as dimensions that we may want to cluster over.</p>
<p>As <a class="reference internal" href="#curse-of-dimensionality"><span class="std std-numref">Fig. 4.1</span></a> illustrates, increasing the number of dimensions that we cluster over makes it exponentially harder for us to identify clusters in the data given a fixed number of individuals in our cohort.
In extreme cases, all samples or individuals look equally distant from each other in the sparse, high-dimensional space.</p>
<p>The curse of dimensionality can be partially reduced by choosing a clustering or outlier detection method which is more robust to the number of dimensions.
However, these still have limits, and in order to overcome these, it is necessary to reduce the number of dimensions in some way, this process is called feature selection.
This can be done through careful curation of important features, through variance cut-offs, or by dimensional reduction methods like Prinipal Component Analysis (PCA) or Multi-dimensional Scaling (MDS) which project the data into a different coordinate system and then discard some of the newly calculated dimensions.</p>
<!--
### Summary
[//]: # (TODO: Big up themes of multi-omics, comibing different sources of data)
-->
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c04-snowflake"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
            <!-- Previous / next buttons -->
<div class='prev-next-area'> 
    <a class='left-prev' id="prev-link" href="0-index.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title"><span class="section-number">4. </span>Phenotype prediction with Snowflake</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="2-snowflake-algorithm.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title"><span class="section-number">4.2. </span><code class="docutils literal notranslate"><span class="pre">Snowflake</span></code> Algorithm</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
        
        </div>
    </div>
    <footer class="footer">
    <div class="container">
      <p>
        
          By Natalie Thurlby<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>
  
  <script src="../_static/js/index.be7d3bbb2ef33a8344ce.js"></script>

  </body>
</html>