
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Abstract &#8212; Phenotype from Genotype</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/custom.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/togglebutton.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script >const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <link rel="shortcut icon" href="../_static/favicon.png"/>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Acknowledgements" href="04-acknowledgements.html" />
    <link rel="prev" title="Declaration" href="02-declaration.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/thesis_logo_with_text.png" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">Phenotype from Genotype</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search the docs ..." aria-label="Search the docs ..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        <p class="caption">
 <span class="caption-text">
  Front Matter
 </span>
</p>
<ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="01-title-page.html">
   <code class="docutils literal notranslate">
   </code>
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="02-declaration.html">
   Declaration
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Abstract
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="04-acknowledgements.html">
   Acknowledgements
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="05-full-table-of-contents.html">
   Table of Contents
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Background
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c01-introduction/intro.html">
   1. Thesis style and philosophy
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c02-biology-background/0-index.html">
   2. How phenotype arises from genotype
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/1-big-questions.html">
     2.1. Big questions: What is genetically determined, and how?
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/2-biological-molecules.html">
     2.2. Biological molecules: DNA, RNA, Proteins and the central dogma of molecular biology.
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/3-more-dna.html">
     2.3. A closer look at DNA: Genomes, Genes, and Genetic Variation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/4-more-proteins.html">
     2.4. Looking more closely at proteins: function, structure and classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/5-phenotype.html">
     2.5. Phenotype
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c02-biology-background/6-summary.html">
     2.6. Summary: how genotype and phenotype are linked
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c03-compbio-background/0-index.html">
   3. How genotype and phenotype are measured and researched
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/1-sequencing-technology.html">
     3.1. Sequencing and microarrays
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/2-measuring-genotype-phenotype.html">
     3.2. From genotype to phenotype: what is measured
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/3-ontologies.html">
     3.3. Ontologies
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/4-comp-bio-methods.html">
     3.4. Predictive computational methods
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/5-bias.html">
     3.5. Sources of bias in computational biology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/6-pqi.html">
     3.6. Proteome Quality Index
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c03-compbio-background/7-summary.html">
     3.7. Summary
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Phenotype prediction
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c04-snowflake/0-index.html">
   4. Phenotype prediction with Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/1-introduction.html">
     4.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/2-snowflake-algorithm.html">
     4.2.
     <code class="docutils literal notranslate">
      <span class="pre">
       Snowflake
      </span>
     </code>
     Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/3-creating-inputs.html">
     4.3. Creating Snowflake inputs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/4-preprocessing.html">
     4.4. Preprocessing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/5-clustering-snps.html">
     4.5. Considerations for Clustering SNPs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c04-snowflake/7-discussion.html">
     4.6. Discussion
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c05-alspac/0-index.html">
   5. Predicting phenotypes of the ALSPAC cohort using Snowflake
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/1-introduction.html">
     5.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c05-alspac/5-discussion.html">
     5.2. Discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Tissue-specific gene expression
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c06-filter/0-index.html">
   6. Filtering computational predictions with tissue-specific expression information
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" type="checkbox"/>
  <label for="toctree-checkbox-5">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/1-introduction.html">
     6.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/2-algorithm.html">
     6.2. Algorithm
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/3-data.html">
     6.3. Data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/5-methods.html">
     6.4. Validation method
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/6-results.html">
     6.5. Filip results
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c06-filter/7-discussion.html">
     6.6. Discussion and Future work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c07-ontolopy/0-index.html">
   7. Ontolopy
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-6" name="toctree-checkbox-6" type="checkbox"/>
  <label for="toctree-checkbox-6">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/1-introduction.html">
     7.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/2-functionality.html">
     7.2. Functionality
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/3-how-it-works.html">
     7.3. Ontolopy tools and practices
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/4-misc-examples.html">
     7.4. Example uses: mapping samples to diseases or phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/5-mapping-example.html">
     7.5. Example use: mapping samples to tissue-related phenotypes
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/6-discussion.html">
     7.6. Discussion
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c07-ontolopy/7-future-work.html">
     7.7. Future Work
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../c08-combining/0-index.html">
   8. Combining RNA-seq datasets
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-7" name="toctree-checkbox-7" type="checkbox"/>
  <label for="toctree-checkbox-7">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/1-background.html">
     8.1. Introduction
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/2-data.html">
     8.2. Data Acquisition
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/3-data-wrangling.html">
     8.3. Data Wrangling
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../c08-combining/7-discussion.html">
     8.4. Results and discussion
    </a>
   </li>
  </ul>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Concluding remarks
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../c09-conclusion/0-conclusion.html">
   9. Concluding remarks
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  End Matter
 </span>
</p>
<ul class="nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/0-appendix.html">
   Appendix
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../cz-end-matter/reference.html">
   Bibliography
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Theme by the <a href="https://ebp.jupyterbook.org">Executable Book Project</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Abstract</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <div class="section" id="abstract">
<h1>Abstract<a class="headerlink" href="#abstract" title="Permalink to this headline">¶</a></h1>
<p>The title of this thesis describes the ambitious scientific aim running through it: explaining the link between genotype and phenotype through molecular biology data.
In our cells, proteins are constantly being created and are degrading.
They are accumulating or interacting to produce the phenotypes that we see at a larger scale: height, levels of enzymes in blood, diseases.
There are hidden molecular explanations for our phenotypes, and our proteins functions.
The types of proteins that it is <em>possible</em> for an organism to produce are determined primarily by its protein-coding DNA.
Meanwhile which of these possible proteins are <em>actively created</em> in each cell are determined by the environment of each cell at each time.
The data about these molecules and their activity is our trail of breadcrumbs in the search for a molecular explanation for phenotype, and these data exist in computational biology’s collection of large databases of community-sourced experimental and computational results.</p>
<p>This thesis explores two main approaches for making and improving explanatory predictions of phenotype and protein function from genotype.
Both predictors seek to leverage the power of the researchers around the world which contribute their results to community databases, and combines these where possible to get a fuller picture of the complex system of interacting molecules.</p>
<p>The first part of this thesis contains all of the necessary background, and contains three chapters.
<a class="reference internal" href="../c01-introduction/intro.html#c01-philosophy"><span class="std std-numref">Chapter 1</span></a> briefly introduces the philosophy of this thesis.
The biology background chapter (<a class="reference internal" href="../c02-biology-background/0-index.html#c02-biology-bg"><span class="std std-numref">chapter 2</span></a>) then presents a detailed overview of the scientific model that links genotype and phenotype.
It tells the story of how phenotype arises from genotype, and introduces the different biological molecules that are involved.
It begins at the very basics: what are DNA, RNA, proteins, and phenotypes; how are they related; how do we categorise them?
This background is intended to make this thesis readable to someone without a background in biology, and to explain the overall aims and context of the research in this thesis.
It does not contain any of my own research.
The computational biology background chapter (<a class="reference internal" href="../c03-compbio-background/0-index.html#c03-compbio-bg"><span class="std std-numref">chapter 3</span></a>) follows on from the previous chapter by discussing popular resources in computational biology, their provenance, and the impact of this on the field.
In this chapter, I also present my contributions to collaborative projects: the Proteome Quality Index paper<span id="id1">[<a class="reference internal" href="../cz-end-matter/reference.html#id9"><span>2</span></a>]</span>, and the 2014 SUPERFAMILY update paper<span id="id2">[<a class="reference internal" href="../cz-end-matter/reference.html#id8"><span>3</span></a>]</span></p>
<p>In the second part, I present the Snowflake phenotype predictor, which uses variants conservation scores, prevalence in the population, and protein domain architectures as input to an unsupervised learning method.
This predictor, the development of which resulted in a patent<span id="id3">[<a class="reference internal" href="../cz-end-matter/reference.html#id10"><span>4</span></a>]</span>, finds unusual combinations of variants associated with phenotypes, and is designed to create explanatory predictions of complex traits.
The algorithm itself, and the results of experiments in validating Snowflake are presented in <a class="reference internal" href="../c04-snowflake/0-index.html#c04-snowflake"><span class="std std-numref">chapter 4</span></a> and <a class="reference internal" href="../c05-alspac/0-index.html#c05-alspac"><span class="std std-numref">chapter 5</span></a>.</p>
<p>In investigating Snowflake’s predictions, it became clear that it was possible for it to include protein-coding SNPs in predictions about phenotypes that exist in tissues in which the protein is never expressed, which brings us to the third and final part of this thesis.
The Filip protein function prediction filter is discussed in <a class="reference internal" href="../c06-filter/0-index.html#c06-filter"><span class="std std-numref">chapter 6</span></a>, which uses gene expression data to filter out predictions of proteins which are not expressed in the tissue relating to a given phenotype.
I discuss attempts to validate Filip’s predictions, including it’s performance in the CAFA3 protein function prediction competition<span id="id4">[<a class="reference internal" href="../cz-end-matter/reference.html#id11"><span>5</span></a>]</span>.
In addition, this part presents tools and datasets that were developed through creating and developing Filip: Ontolopy a Python package for querying OBO files in <a class="reference internal" href="../c07-ontolopy/0-index.html#c06-ontolopy"><span class="std std-numref">chapter 7</span></a>, and a combined data set of gene expression data in <a class="reference internal" href="../c08-combining/0-index.html#c05-combining"><span class="std std-numref">chapter 8</span></a>.</p>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./c0-front-matter"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="02-declaration.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Declaration</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="04-acknowledgements.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Acknowledgements</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Natalie Zelenka<br/>
  
      &copy; Copyright 2020.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>